spec/services/spam_and_bounce_resets_spec.rb

```diff
@@ -0,0 +1,86 @@
+require 'rails_helper'
+
+RSpec.describe SpamAndBounceResets do
+  let(:visit) { build(:visit) }
+  subject { described_class.new(visit) }
+
+  describe '.perform_resets' do
+    context 'override_email_check == true' do
```

Comment:
Use something like 'when' to start contexts so that they form comprehensible names when glued together. (Personally, I'd prefer English like 'is true' or 'is false' to '== true', as well.)